lib/ecstatic.js

```diff
@@ -124,13 +124,26 @@ var ecstatic = module.exports = function (dir, options) {
               url: parsed.pathname + '.' + defaultExt + ((parsed.search)? parsed.search:'')
             }, res, next);
           }
-          else {
-            // Try to serve default ./404.html
+          else if (!handleError) {
+            // If we're not handling errors/fallbacks, bail out now
+            middleware({
+              url: req.url,
+              statusCode: 404
+            }, res, next);
+          }
+          else if (path.basename(parsed.pathname, defaultExt) === '200.') {
+            // Already tried ./200.html, now try ./404.html
             middleware({
-              url: (handleError ? ('/' + path.join(baseDir, '404.' + defaultExt)) : req.url),
+              url: '/' + path.join(baseDir, '404.' + defaultExt),
               statusCode: 404
             }, res, next);
           }
+          else {
+            // Try ./200.html before falling back to ./404.html
```

Comment:
It's mildly surprising that this isn't opt-in behavior, but 200.html wasn't used for anything else prior, right? So this is a breaking change, but that seems fine.